kaldb/src/main/java/com/slack/kaldb/server/RecoveryTaskFactory.java

```diff
@@ -0,0 +1,81 @@
+package com.slack.kaldb.server;
+
+import com.slack.kaldb.metadata.recovery.RecoveryTaskMetadata;
+import com.slack.kaldb.metadata.recovery.RecoveryTaskMetadataStore;
+import com.slack.kaldb.metadata.snapshot.SnapshotMetadata;
+import com.slack.kaldb.metadata.snapshot.SnapshotMetadataStore;
+import com.slack.kaldb.util.SnapshotsUtil;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is responsible for the indexer startup operations like stale live snapshot cleanup.
+ * determining the start indexing offset and create Recovery task etc..
+ */
+public class RecoveryTaskFactory {
```

Comment:
Instead of `RecoveryTaskFactory` this might be better as something like a `StartupOperationsUtils`. This class doesn't implement a factory pattern, and we already have expanded this to include non-recovery tasks, such as "stale live snapshot" cleanup.

Reply:
I renamed this class `StartupOperationsUtils` but that was even more vague than `RecoveryTaskFactory`. So, left the name as is. Let's change the name when we integrate this code with the indexer.